homeassistant/components/hydrawise/switch.py

```diff
@@ -21,16 +23,37 @@
 from .coordinator import HydrawiseDataUpdateCoordinator
 from .entity import HydrawiseEntity
 
-SWITCH_TYPES: tuple[SwitchEntityDescription, ...] = (
-    SwitchEntityDescription(
+
+@dataclass(frozen=True, kw_only=True)
+class HydrawiseSwitchEntityDescription(SwitchEntityDescription):
+    """Describes Hydrawise binary sensor."""
+
+    turn_on_fn: Callable[[HydrawiseSwitch], Coroutine[Any, Any, None]]
+    turn_off_fn: Callable[[HydrawiseSwitch], Coroutine[Any, Any, None]]
```

Comment:
Since all actions are done to self.coordinator.api, you could also just use the API object as input

Reply:
We still need the zone if from the switch.

Reply:
You could add that as separate argument

Reply:
Ok.